### PR TITLE
[W4P] Limits, Learning and Platform

### PR DESCRIPTION
**Pull request by deadlypants1973**

### CODE REVIEW

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Workers for Platforms is built on top of [Cloudflare Workers](/workers/). The same security and performance models apply to applications that use Workers for Platforms. Workers for Platforms extends the capabilities of Workers for SaaS businesses that want to deploy Worker scripts on behalf of their customers or that want to let their users write Worker scripts directly.
+
+## Architecture
+
+Workers for Platforms introduces a new architecture model as outlined on this page.
+
+### Dispatch namespace
+
+A dispatch namespace is composed of a collection of user Workers. With dispatch namespaces, a dynamic dispatch Worker can be used to call any User Worker in a namespace. Namespaces can be used to ____.
```
Comment (deadlypants1973) [inline content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]: @tanushree-sharma this was cutoff in the google doc

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Below you will find an example request lifecycle in the Workers for Platforms architecture.
+
+![The request lifecycle is described below.](../media/WFP-diagram.png)
+
+In the above diagram:
+
+1. Request for `customer-a.example.com/api` will first hit the dynamic dispatch Worker (`api-prod`).
+2. The dispatcher (`env.dispatcher.get(customer-a)`) will handle routing logic to user Workers based on properties of the request.
+3. The subdomain (`customer-a.example.com`) of the incoming request is used to route to the user Worker with the same name (`customer-a`).
```
Comment (deadlypants1973) [inline content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]: @tanushree-sharma I rewrote some of this, can you check for accuracy?

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+If you have multiple distinct services you are providing your customers, you should split these out into different dispatch workers and namespaces. We discourage creating a new namespace for each customer.
+
+{{</Aside>}}
+
+### Dynamic dispatch Worker
+
+A dynamic dispatch Worker is written by Cloudflare’s platform customers to run their own logic before dispatching (routing) the request to user Workers. In addition to routing, it can be used to run authentication, create boilerplate functions and sanitize responses.
+
+The dynamic dispatch Worker calls user Workers from the dispatch namespace and executes them. The dynamic dispatch Worker is configured with a dispatch namespace binding. The binding is the entrypoint for all requests to user Workers. Refer to
```
Comment (marciocloudflare) [inline content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]: `Refer to` --> This is missing content

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+### Dynamic dispatch Worker
+
+A dynamic dispatch Worker is written by Cloudflare’s platform customers to run their own logic before dispatching (routing) the request to user Workers. In addition to routing, it can be used to run authentication, create boilerplate functions and sanitize responses.
+
+The dynamic dispatch Worker calls user Workers from the dispatch namespace and executes them. The dynamic dispatch Worker is configured with a dispatch namespace binding. The binding is the entrypoint for all requests to user Workers. Refer to 
+
+### User Workers
+
+User Workers are written by your end users (end developers). End developers deploy user Workers to script automated actions, create integrations or modify response payload to return custom content.
```
Comment (marciocloudflare) [inline content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]:
```suggestion
User Workers are written by your end users (end developers). End developers deploy user Workers to script automated actions, create integrations or modify response payloads to return custom content.
```

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+![The request lifecycle is described below.](../media/WFP-diagram.png)
+
+In the above diagram:
+
+1. Request for `customer-a.example.com/api` will first hit the dynamic dispatch Worker (`api-prod`).
+2. The dispatcher (`env.dispatcher.get(customer-a)`) will handle routing logic to user Workers based on properties of the request.
+3. The subdomain (`customer-a.example.com`) of the incoming request is used to route to the user Worker with the same name (`customer-a`).
+
+## ​Workers for Platforms versus Service Bindings
```
Comment (marciocloudflare) [inline content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]:
```suggestion
## ​Workers for Platforms versus Service bindings
```

Comment (marciocloudflare) [reply content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]: I lowercased the `b` as in all other instances this is lowercase

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+Service bindings explicitly link two Workers together. They are meant for use cases where you know exactly which Workers need to communicate with each other. Service bindings do not work in the Workers for Platforms model because user Workers are uploaded as needed by your end users.
+
+In the Workers for Platforms model, a dynamic dispatch Worker can be used to call any user Worker (similar to how Service bindings work) in a dispatch namespace but without needing to explicitly pre-define the relationship.
+
+Service bindings and Workers for Platforms can be used simultaneously when building applications. 
+
+
+
+
```
Comment (marciocloudflare) [inline content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]:
```suggestion
```

File: content/cloudflare-for-platforms/workers-for-platforms/learning/_index.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+pcx_content_type: concept
+title: Learning
+weight: 8
+meta:
+    title: Learning
+---
+
+{{<directory-listing>}}
```
Comment (marciocloudflare) [inline content/cloudflare-for-platforms/workers-for-platforms/learning/_index.md:9]: Will this folder have more content inside in the future? If not consider changing this folder into the `learning.md` page

File: content/cloudflare-for-platforms/workers-for-platforms/platform/limits.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+​​
+## Bindings
+
+You can use Workers [bindings](/workers/platform/bindings) with the dynamic dispatch Worker or any namespaced Workers. Your user Workers cannot be defined as Durable Objects. User Workers can have a Durable Objects binding.  
+
+Bindings for your dynamic dispatch Worker or any user Workers can be defined on multipart script uploads in the metadata blob.
+
+{{<Aside type="warning">}}
+
+Any product specific limits still apply to your account (for example, the [100 KV namespace limit](/workers/platform/limits/#kv-limits)). You can request adjustments to limits that conflict with your project goals by contacting Cloudflare. To increase a limit, complete the [Limit Increase Request Form](https://forms.gle/ukpeZVLWLnKeixDu7).
```
Comment (marciocloudflare) [inline content/cloudflare-for-platforms/workers-for-platforms/platform/limits.md]:
```suggestion
Any product-specific limits still apply to your account (for example, the [100 KV namespace limit](/workers/platform/limits/#kv-limits)). You can request adjustments to limits that conflict with your project goals by contacting Cloudflare. To increase a limit, complete the [Limit Increase Request Form](https://forms.gle/ukpeZVLWLnKeixDu7).
```

**Review by marciocloudflare — CHANGES_REQUESTED** [review]

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md (hunk 1 above)
Comment (tanushree-sharma) [reply content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]:
```suggestion
A dispatch namespace is composed of a collection of user Workers. With dispatch namespaces, a dynamic dispatch Worker can be used to call any User Worker in a namespace.
```

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md (hunk 2 above)
Comment (tanushree-sharma) [reply content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]:
```suggestion
1. Request for `customer-a.example.com/api` will first hit the dynamic dispatch Worker (`api-prod`).
2. The dispatcher (`env.dispatcher.get(customer-a)`) configured in your dynamic dispatch Worker code will handle routing logic to user Workers.
3. The subdomain (`customer-a.example.com`) of the incoming request is used to route to the user Worker with the same name (`customer-a`).
```

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
```
Comment (deadlypants1973) [inline content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]:
```suggestion
```

File: content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+If you have multiple distinct services you are providing your customers, you should split these out into different dispatch workers and namespaces. We discourage creating a new namespace for each customer.
+
+{{</Aside>}}
+
+### Dynamic dispatch Worker
+
+A dynamic dispatch Worker is written by Cloudflare’s platform customers to run their own logic before dispatching (routing) the request to user Workers. In addition to routing, it can be used to run authentication, create boilerplate functions and sanitize responses.
+
+The dynamic dispatch Worker calls user Workers from the dispatch namespace and executes them. The dynamic dispatch Worker is configured with a dispatch namespace binding. The binding is the entrypoint for all requests to user Workers. Refer to
```
Comment (deadlypants1973) [inline content/cloudflare-for-platforms/workers-for-platforms/learning/how-workers-for-platforms-works.md]:
```suggestion
The dynamic dispatch Worker calls user Workers from the dispatch namespace and executes them. The dynamic dispatch Worker is configured with a [dispatch namespace binding](/workers/platform/bindings/#dispatch-namespace-bindings-workers-for-platforms). The binding is the entrypoint for all requests to user Workers.
```

**Review by marciocloudflare — APPROVED** [review]

**Pull request merged**